src/server/game/Entities/Player/Player.cpp

```diff
@@ -5638,25 +5648,24 @@ void Player::SetSkill(uint16 id, uint16 step, uint16 newVal, uint16 maxVal)
                 SetUpdateFieldValue(m_values.ModifyValue(&Player::m_activePlayerData).ModifyValue(&UF::ActivePlayerData::ProfessionSkillLine, 1), 0);
         }
     }
-    else                            //add
+    else
     {
-        // Check if the player already has a skill, otherwise pick a empty skill slot if available
-        uint8 skillSlot = itr != mSkillStatus.end() ? itr->second.pos : 0;
-        if (!skillSlot)
+        // We are about to learn a skill that has been added outside of normal circumstances (Game Master command, scripts etc.)
+        uint8 skillSlot = 0;
+
+        // Find a free skill slot
+        for (uint32 i = 0; i < PLAYER_MAX_SKILLS; ++i)
         {
-            for (uint32 i = 0; i < PLAYER_MAX_SKILLS; ++i)
+            if (!m_activePlayerData->Skill->SkillLineID[i])
             {
-                if (!m_activePlayerData->Skill->SkillLineID[i])
-                {
-                    skillSlot = i;
-                    break;
-                }
+                skillSlot = i;
+                break;
             }
         }
 
         if (!skillSlot)
         {
-            TC_LOG_ERROR("misc", "Tried to add skill #%u but the player cannot have additional skills", id);
+            TC_LOG_ERROR("misc", "Tried to add skill %u but player %s (%s) cannot have additional skills", id, GetName().c_str(), GetGUID().ToString().c_str());
             return;
         }
```

Comment:
we have remove this code and the server doesnt not crash after

Reply:
Dude, would you kindly give me a proper crashlog already?

Reply:
it does not want to work currently it says that there are missing files in debug mode

Reply:
so basically @buddyqc69 meant he cant compile core in debug to get proper crashlog.

Reply:
Somehow I have a feeling that he isn't even using the official TC branch. I would like to get a test result on a actual TC version so I can merge this thing at last.

Reply:
We should do some cmake magic to disable release build targets :p

Reply:
nope i use the latest trinitycore :)